.github/workflows/build.yaml

```diff
@@ -39,46 +39,40 @@ jobs:
         run: echo "tag=${GITHUB_REF#refs/*/}" >> $GITHUB_OUTPUT
       - name: conda setup
         run: |
-          eval "$(conda shell.bash hook)"
-          conda config --set always_yes yes --set changeps1 no
-          conda update conda
+          conda update --name base conda
           conda install anaconda-client conda-build
+          pip install hatch
       - name: conda build
         run: |
-          eval "$(conda shell.bash hook)"
-          conda build conda.recipe/
+          VERSION=`hatch version` conda build conda.recipe/
       - name: conda dev upload
         if: (github.event_name == 'push' && (contains(steps.vars.outputs.tag, 'a') || contains(steps.vars.outputs.tag, 'b') || contains(steps.vars.outputs.tag, 'rc')))
         run: |
-          eval "$(conda shell.bash hook)"
-          anaconda --token $CONDA_UPLOAD_TOKEN upload --user pyviz --label=dev $(conda build --output conda.recipe)
+          anaconda --token $CONDA_UPLOAD_TOKEN upload --user pyviz --label=dev $(VERSION=`hatch version` conda build --output conda.recipe)
       - name: conda main upload
         if: (github.event_name == 'push' && !(contains(steps.vars.outputs.tag, 'a') || contains(steps.vars.outputs.tag, 'b') || contains(steps.vars.outputs.tag, 'rc')))
         run: |
-          eval "$(conda shell.bash hook)"
-          anaconda --token $CONDA_UPLOAD_TOKEN upload --user pyviz --label=dev --label=main $(conda build --output conda.recipe)
+          anaconda --token $CONDA_UPLOAD_TOKEN upload --user pyviz --label=dev --label=main $(VERSION=`hatch version` conda build --output conda.recipe)
   pip_build:
     name: Build PyPI Packages
     runs-on: 'ubuntu-latest'
     defaults:
       run:
         shell: bash -l {0}
-    env:
-      TOX_ENV: "py3.7"
     steps:
       - uses: actions/checkout@v3
       - name: Fetch unshallow
         run: git fetch --prune --tags --unshallow -f
       - uses: actions/setup-python@v4
         with:
-          python-version: "3.7"
+          python-version: "3.9"
       - name: env setup
         run: |
           python -m pip install --upgrade pip
-          python -m pip install setuptools wheel twine tox
+          python -m pip install build
       - name: pip build
         run: |
-          python setup.py sdist bdist_wheel
+          python -m build
```

Comment:
Did you try `hatch build`?

Reply:
Yes, both with the same result. I saw `hatch` is using `build` in its CI (https://github.com/pypa/hatch/blob/master/.github/workflows/build-hatch.yml#L23). Also, I'm not sure whether `hatch build` can build the package if another backend is used, while I believe `python -m build` will build whatever the backend.

Reply:
It could be a chicken and egg situation. 

I was mostly curious about it :+1: